add-ons/VoicemailTutorial/VoicemailEkataPhoneValid.js

```diff
@@ -0,0 +1,47 @@
+// This code sample serves to filter robocalls prior to taking a voicemail.
+
+// Twilio Functions automatically executes the exports.handler method in any
+// code you write. It comes with inputs for a context, event, and callback.
+// Full documentation: https://www.twilio.com/docs/runtime/functions/invocation
+exports.handler = (context, event, callback) => {
+    // Create a new voice response object
+    const twiml = new Twilio.twiml.VoiceResponse();
+    const addOns = event.AddOns;
+    let blockCalls = false;
+    
+    // verify that the addOn executed properly
+    if (addOns.status === 'successful') {
+      // if the addOn executed properly, verify that the number is legitimate.
+      blockCalls = ekata_results(addOns.results.ekata_phone_valid);
+    }
```

Comment:
I might be missing context here, but perhaps...

```suggestion
    // If the addOn executed properly, verify that the number is legitimate.
    const blockCalls = addOns.status === 'successful'
      ? ekataResults(addOns.results.ekata_phone_valid)
      : false;
```

or

```suggestion
  const blockCalls = ekataResults(addOns);
  
  ...
  
  function ekataResults(addOns) {
    // Early return
    if (addOns.status !== 'successful') return false;
    
    const ekata = addOns.results.ekata_phone_valid;
    // Bonus early return for no valid results or non-successful status
    if (!ekata || ekata.status !== "success") return false; 
    
    return ekata.result.is_valid;
}
```